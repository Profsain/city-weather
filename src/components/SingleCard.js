import PropTypes from 'prop-types';
import React from 'react';
import '../styles/SingleCard.css';

const SingleCard = ({
  temp,
  humidity,
  condition,
  icon,
}) => (

  <div className="Card">
    <h4>{condition}</h4>
    <img src={icon} alt="weather icon" />
    <div className="Report">
      <p>
        {temp}
        â„ƒ
      </p>
      <p>
        {humidity}
        %
      </p>
    </div>
  </div>
);

SingleCard.propTypes = {
  condition: PropTypes.string,
  humidity: PropTypes.number,
  icon: PropTypes.string,
  temp: PropTypes.number,
};

export default SingleCard;
