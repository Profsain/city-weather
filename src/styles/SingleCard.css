.Card {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  background-color: #1e213a;
  width: 140px;
  font-size: 18px;
}

.Card img {
  width: 80px;
  height: 60px;
}

.Report {
  display: flex;
  justify-content: space-between;
}

.Report p {
  margin: 16px;
}
