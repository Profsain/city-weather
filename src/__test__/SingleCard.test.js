import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../redux/configurestore';
import SingleCard from '../components/SingleCard';

it('Renders SingleCard component correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <SingleCard />
    </Provider>,
  );
  expect(tree).toMatchSnapshot();
});
