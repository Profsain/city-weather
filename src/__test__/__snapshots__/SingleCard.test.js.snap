// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders SingleCard component correctly 1`] = `
<div
  className="Card"
>
  <h4 />
  <img
    alt="weather icon"
  />
  <div
    className="Report"
  >
    <p>
      â„ƒ
    </p>
    <p>
      %
    </p>
  </div>
</div>
`;
